File: src/app/shared/dto/Patient.ts
/**
 *
 * Informations about the patient.
 * @export
 * @interface Patient
 */
export interface Patient {
  /**
   * Id of the patient on DB.
   * @type {number}
   * @memberof Patient
   */
  id?: number,
  /**
   * First name of the patient.
   *
   * @type {string}
   * @memberof Patient
   */
  firstName: string,
  /**
   * Last name of the patient.
   *
   * @type {string}
   * @memberof Patient
   */
  lastName: string,
  /**
   * Mail address of the patient.
   *
   * @type {string}
   * @memberof Patient
   */
  mail: string
}

File: src/main/java/fr/polytech/covid/entity/Patient.java
package fr.polytech.covid.entity;

import javax.persistence.*;

@Entity
@Table(name = "patient")
public class Patient {
    @Id
    @GeneratedValue
    private int id;
    private String firstName;
    private String lastName;
    private String mail;
    private boolean isVaccinated;


    public Patient() {
    }

    public Patient(String firstName, String lastName, String mail) {
        this.firstName = firstName;
        this.lastName = lastName;
        this.mail = mail;
        this.isVaccinated = false;
    }

    public int getId() {
        return id;
    }

    public String getFirstName() {
        return firstName;
    }

    public void setFirstName(String firstName) {
        this.firstName = firstName;
    }

    public String getLastName() {
        return lastName;
    }

    public void setLastName(String lastName) {
        this.lastName = lastName;
    }

    public String getMail() {
        return mail;
    }

    public void setMail(String mail) {
        this.mail = mail;
    }

    public boolean isVaccinated() {
        return isVaccinated;
    }

    public void setVaccinated(boolean vaccinated) {
        isVaccinated = vaccinated;
    }


}
